_source/_posts/2022-02-21-build-deploy-node-app-heroku.md

```diff
@@ -0,0 +1,335 @@
+---
+layout: blog_post
+title: "Build and Deploy a Node.js App to Heroku"
+author: nickolas-fisher
+by: contractor
+communities: [javascript]
+description: "A tutorial that shows you how to build and deploy a Node.js app to Heroku."
+tags: [javascript, node, heroku, express]
+tweets:
+- "Learn how to build and deploy a @nodejs app to @heroku 🔥"
+image: blog/build-deploy-node-app-heroku/cover-image.jpg
+type: conversion
+github: https://github.com/oktadev/okta-node-heroku-app-example
+---
+
+Heroku is a platform as a service that supports many languages.  Initially, it supported only Ruby sites but now supports various languages, including JavaScript with Node.js.  Heroku also has docker support so that you can deploy just about anything to it.  
+
+This tutorial will teach you how to build a small application using the Express framework for Node.js.  You will then secure that application using Okta by creating an application in Okta for web applications and integrating the Okta OIDC middleware with your application.  Finally, you will learn how to deploy the application to Heroku, update your Okta application for your Heroku site, and set the environment variables for your app in Heroku.
```

Comment:
"You will then secure that application using Okta by creating an application in Okta for web applications and integrating the Okta OIDC middleware with your application." - This sentence uses the word application 4 times, and I don't understand what's going on - could we say: 
You will then secure your Express app with Okta, by creating a web version in Okta and integrating the Okta OIDC middleware with your original app." - does this make sense or have I changed the meaning?